Sources/Sentry/SentrySwizzle.m

```diff
@@ -40,17 +41,104 @@ - (SentrySwizzleOriginalIMP)getOriginalImplementation
 
 @implementation SentrySwizzle
 
+// This lock is shared by all swizzling and unswizzling calls to ensure that
+// only one thread is modifying the class at a time.
+static pthread_mutex_t gLock = PTHREAD_MUTEX_INITIALIZER;
+
+#if TEST || TESTCI
+/**
+ * - Returns: a dictionary that maps keys to the references to the original implementations.
+ */
+static NSMutableDictionary<NSValue *, NSValue *> *
+refsToOriginalImplementationsDictionary(void)
+{
+    static NSMutableDictionary *refsToOriginalImplementations;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ refsToOriginalImplementations = [NSMutableDictionary new]; });
+    return refsToOriginalImplementations;
+}
+
+/**
+ * Adds a reference to the original implementation to the dictionary.
+ *
+ * If the key is NULL, it will log an error and NOT store the reference.
+ *
+ * - Parameter key: The key for which to store the reference to the original implementation.
+ * - Parameter implementation: Reference to the original implementation to store.
+ */
```

Comment:
`l`: Thanks for the code docs, but I don't think you have to add these for internal methods. We can keep them, but you can safe some extra work next time if you want.

Reply:
I added the comments initially to explain the pre-condition that key must not be NULL, but then decided to add null-handling instead.